tests/run-fedora

```diff
@@ -19,12 +19,17 @@ mkdir -p /run/systemd/system
 
 # run build and test as user
 useradd build
-su -s /bin/sh - build << EOF
+su -s /bin/sh - build << EOF || { [ -z "$DEBUG" ] || sleep infinity; exit 1; }
 set -ex
 cd /source
 export TEST_CODE="$TEST_CODE"
-python3 -m unittest -v || {
-  [ -z "$DEBUG" ] || sleep infinity
-  exit 1
-}
+
+python3 -m unittest -v
```

Comment:
How about something like this:
```sh
case "USE_SUITE" in
   unittest)
       RUN_UNITEST=1
       ;;
   pytest)
       RUN_PYTEST=1
       ;;
   *)
       RUN_PYTEST=1
       RUN_UNITTEST=1
       ;;
esac
if [ -n "$RUN_UNITEST" ]; then
   python3 -m unittest -v
fi
if [ "$RUN_PYTEST" ]; then
   python3 -m pytest -v
fi
```
and then in the github workload you can 
```yaml
strategy:
      matrix:
        USE_SUITE: ["unittest", "pytest"]
```
to get the jobs to split up nicely - the default behaviour of the script is to run both.

Reply:
TBH I think this is a bit overkill. I dropped the TEST_CODE dependency, and just run everything everywhere. It's not that expensive, and it's good to know when e.g. pytest fails in rawhide.